src/phases/execution/LookupCommands.yml

```diff
@@ -59,6 +59,16 @@ ForeignSmallDocTemplate:
   arr_key: {^Array: {of: *RandIntUpTo, number: 5}}
   data1: *Rand32BitInt
 
+# The document template for both local and foreign collections in NLJ Lookups. The doc contains
+# scalar, single element array, 8 elements array, simple object and nested object.
+LargeComplexDocTemplate:
+  int_key1: &RandInt0To1k {^RandomInt: {min: 0, max: 1000}}
+  int_key2: {^RandomInt: {min: 2000, max: 3000}}
```

Comment:
Should this be from 10000 to 12000:
```
r4: Random.randInt(10000) + 2000,
````
?

This might not matter however, since my understanding of r4 is that it's a value that's not meant to match any of the others (so as to stress the join logic, and not the logic to materialize documents).

Speaking of: in addition to the tests from https://docs.google.com/document/d/1XGMbzMrIrY1JX7uWOIbKjcdJMlQfu5cdcxtnIXVJ3gY/edit#heading=h.if6c26hl7ubr do you know if we have tests that test NLJ that produce matches? i.e. do you know if https://github.com/10gen/PerformanceReports/blob/main/scripts/tags.yml#L990-L994 produce a non-empty "as" array? This would be good to have in order to verify that the cost of materializing documents isn't a problem in SBE.

Reply:
oh, you are elaborative , I didn't find this after reading it many times.
For your second question, `Foreign9500_IntTopField` query will for sure to produce non-empty result, other queries could have non-empty result depends on random number.

Reply:
Ack; resolving.